Apps/WebClient/src/ClientApp/src/components/navmenu/HeaderComponent.vue

```diff
@@ -24,285 +23,266 @@ import { ILogger } from "@/services/interfaces";
-                new DateWrapper(n.scheduledDateTimeUtc, {
-                    isUtc: true,
-                    hasTime: true,
-                }).isAfter(lastLoginDateTime)
-            );
+function onScroll(): void {
+    const st = window.scrollY || document.documentElement.scrollTop;
+    if (
+        Math.abs(st - lastScrollTop.value) > minimumScrollChange &&
+        isMobileWidth.value
+    ) {
+        if (st > lastScrollTop.value) {
+            // down-scroll
+            if (isHeaderShown.value) {
+                setHeaderState(false);
+            }
+        } else {
+            // up-scroll
+            if (!isHeaderShown.value) {
+                setHeaderState(true);
+            }
         }
-        return this.notifications;
     }
+    // For Mobile or negative scrolling
+    lastScrollTop.value = st <= 0 ? 0 : st;
+}
 
-    private get notificationBadgeContent(): string | boolean {
-        const count = this.newNotifications.length;
-        return count === 0 || this.notificationButtonClicked
-            ? false
-            : count.toString();
-    }
+function handleToggleClick(): void {
+    toggleSidebar();
+}
 
-    private get highlightTourChangeIndicator(): boolean {
-        return this.user.hasTourUpdated && !this.hasViewedTour;
+function handleLogoutClick(): void {
+    if (isValidIdentityProvider.value) {
+        showRating();
+    } else {
+        processLogout();
     }
+}
 
-    private onScroll(): void {
-        const st = window.scrollY || document.documentElement.scrollTop;
-        if (
-            Math.abs(st - this.lastScrollTop) >
-                HeaderComponent.minimumScrollChange &&
-            this.isMobileWidth
-        ) {
-            if (st > this.lastScrollTop) {
-                // downscroll
-                if (this.isHeaderShown) {
-                    this.setHeaderState(false);
-                }
-            } else {
-                // upscroll
-                if (!this.isHeaderShown) {
-                    this.setHeaderState(true);
-                }
-            }
-        }
-        // For Mobile or negative scrolling
-        this.lastScrollTop = st <= 0 ? 0 : st;
-    }
+function handleShowTourClick(): void {
+    hasViewedTour.value = true;
+    appTourComponent.value?.showModal();
+}
 
-    private handleToggleClick(): void {
-        this.toggleSidebar();
-    }
+function showRating(): void {
+    ratingComponent.value?.showModal();
+}
 
-    private toggleMenu(): void {
-        this.toggleSidebar();
-    }
+function processLogout(): void {
+    logger.debug(`redirecting to logout view ...`);
+    router.push({ path: "/logout" });
+}
 
-    private handleLogoutClick(): void {
-        if (this.isValidIdentityProvider) {
-            this.showRating();
-        } else {
-            this.processLogout();
-        }
+watch(isMobileWidth, (isMobileWidth) => {
+    if (!isMobileWidth) {
+        setHeaderState(false);
     }
+});
```

Comment:
```suggestion
watch(isMobileWidth, (value) => {
    if (!value) {
        setHeaderState(false);
    }
});
```

Avoids using the same name for the local variable.